git/helpers.sh

```diff
@@ -0,0 +1,59 @@
+#
+# Deletes a branch locally and remotely
+#
+function gitdelete() {
+  if [ -n "$1" ]; then
+    local_exists=`git show-ref refs/heads/$1`
+    remote_exists=`git ls-remote origin $1`
+    if [ -n "$local_exists" ]; then
+      git branch -D $1;
+    else
+      echo "could not find local branch: $1";
+    fi
+    if [ -n "$remote_exists" ]; then
+      git ps origin :$1;
+    else
+      echo "could not find remote branch: $1";
+    fi
+  fi
+}
+
+#
+# Merges a remote source branch into a remote destination branch (that defaults to `develop`)
+# and deletes the source branch locally and remotely
+#
+function gitmerge() {
+  SOURCE_BRANCH="$1"
+  DESTINATION_BRANCH="${2:-develop}"
+  echo "merging $SOURCE_BRANCH into $DESTINATION_BRANCH..."
+  if [[ $SOURCE_BRANCH == "" ]]; then
+    echo "Please provide the name of the branch to be used"
+    exit
+  fi
+  git fetch origin $SOURCE_BRANCH && \
+  git stash && \
+  git checkout $DESTINATION_BRANCH && \
+  git pull origin $DESTINATION_BRANCH && \
+  git merge origin/$SOURCE_BRANCH
+
+  RETURN_CODE="$?"
+  if [[ $RETURN_CODE == "0" ]]; then
+    git push origin $DESTINATION_BRANCH && \
+    gitdelete $SOURCE_BRANCH
+  else
+    echo "Operation aborted with return code $RETURN_CODE"
+  fi
+}
+
+#
+# Creates a release branch from an up-to-date `develop` branch and pushed it to remote
+# Note: uncommited changes are stashed for you to recover later
+#
+function gitrelease() {
+  git stash && \
```

Comment:
We could add `git fetch origin develop && \` here

Reply:
That's not necessary... We have `git pull origin develop` to ensure that the branch is up-to-date (I prefer `pull` over `fetch` in this case).